contracts/GeneralizedSwapMigrator.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./interfaces/ISwap.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title GeneralizedSwapMigrator
+ * @notice This contract is responsible for migration liquidity between pools
+ * Users can use this contract to remove their liquidity from the old pools and add them to the new
+ * ones with a single transaction.
+ */
+contract GeneralizedSwapMigrator is Ownable {
+    using SafeERC20 for IERC20;
+
+    struct MigrationData {
+        IERC20 oldPoolLPTokenAddress;
+        IERC20 newPoolLPTokenAddress;
+        IERC20[] underlyingTokens;
+    }
+
+    uint256 private constant MAX_UINT256 = 2**256 - 1;
+    mapping(address => mapping(address => MigrationData)) migrationMap;
+
+    constructor() public Ownable() {}
+
+    /**
+     * @notice Add new migration data to the contract
+     * @param oldPoolAddress pool address to migrate from
+     * @param newPoolAddress pool address to migrate to
+     * @param mData MigrationData struct that contains information of the old and new pools
+     * @param overwrite should overwrite existing migration data
+     */
+    function addMigrationData(
+        address oldPoolAddress,
+        address newPoolAddress,
+        MigrationData memory mData,
+        bool overwrite
+    ) external onlyOwner {
+        // Check
+        if (!overwrite) {
+            require(
+                address(
+                    migrationMap[oldPoolAddress][newPoolAddress]
+                        .oldPoolLPTokenAddress
+                ) != address(0),
+                "cannot overwrite migrationData"
+            );
+        }
+        require(
+            address(mData.oldPoolLPTokenAddress) != address(0),
+            "oldPoolLPTokenAddress == 0"
+        );
+        require(
+            address(mData.newPoolLPTokenAddress) != address(0),
+            "newPoolLPTokenAddress == 0"
+        );
+
+        // Effect
+        migrationMap[oldPoolAddress][newPoolAddress] = mData;
```

Comment:
Would we consider updating any other entries that listed `oldPoolAddress` as a destination to now point to `newPoolAddress`? I guess that would run into gaslimit issues